File: packages/core/commands/prune.ts
import { Command } from 'commander';
import { SimpleGit } from 'simple-git';
import { AbstractCommand } from './base';

export class PruneCommand extends AbstractCommand {
  constructor(protected git: SimpleGit) {
    super(git);
  }

  public load(program: Command): void {
    program
      .command('prune')
      .description('Prune stale worktree information')
      .option('--dry-run', 'Do not remove anything; show what would be pruned')
      .option('--verbose', 'Report all removals')
      .option('--expire <time>', 'Expire worktrees older than <time>')
      .action(async (options: { dryRun?: boolean; verbose?: boolean; expire?: string }) => {
        await this.pruneWorktrees(options);
      });
  }

  private async pruneWorktrees(options: { dryRun?: boolean; verbose?: boolean; expire?: string }): Promise<void> {
    const args = ['worktree', 'prune'];
    if (options.dryRun) args.push('--dry-run');
    if (options.verbose) args.push('--verbose');
    if (options.expire) args.push(`--expire=${options.expire}`);

    const out = await this.git.raw(args);
    if (out) process.stdout.write(out);
  }
}

File: src/cmd/prune.rs
use anyhow::Result;
use clap::Args;
use wt_core::git::GitRepo;
use wt_core::types::PruneOptions;
use wt_core::worktree;

#[derive(Args, Debug)]
pub struct PruneArgs {
    /// Do not remove anything; show what would be pruned
    #[arg(long)]
    dry_run: bool,

    /// Report all removals
    #[arg(long)]
    verbose: bool,

    /// Expire worktrees older than <time>
    #[arg(long)]
    expire: Option<String>,
}

pub fn run(args: PruneArgs) -> Result<()> {
    let repo = GitRepo::open(&std::env::current_dir()?)?;
    let opts = PruneOptions {
        dry_run: args.dry_run,
        verbose: args.verbose,
        expire: args.expire,
    };
    let output = worktree::prune_worktrees(&repo, opts)?;
    if !output.is_empty() {
        print!("{}", output);
    }
    Ok(())
}
